File: backend/app/models/mission.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Enum, Text, ForeignKey, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database.base import Base
import enum

class MissionPriority(str, enum.Enum):
    CRITIQUE = "critique"
    URGENTE = "urgente"
    NORMALE = "normale"
    FAIBLE = "faible"

class MissionStatus(str, enum.Enum):
    EN_ATTENTE = "en_attente"
    ASSIGNEE = "assignee"
    EN_COURS = "en_cours"
    TERMINEE = "terminee"
    ANNULEE = "annulee"

class Mission(Base):
    __tablename__ = "missions"

    id = Column(Integer, primary_key=True, index=True)
    patient_name = Column(String(100), nullable=False)
    patient_phone = Column(String(20), nullable=False)
    patient_age = Column(Integer)
    patient_condition = Column(String(200), nullable=False)
    priority = Column(Enum(MissionPriority), nullable=False)
    status = Column(Enum(MissionStatus), default=MissionStatus.EN_ATTENTE)
    
    # Localisation de prise en charge
    pickup_address = Column(String(500), nullable=False)
    pickup_latitude = Column(Float, nullable=False)
    pickup_longitude = Column(Float, nullable=False)
    
    # Destination
    hospital_id = Column(Integer, ForeignKey("hospitals.id"), nullable=False)
    
    # Assignation
    ambulance_id = Column(Integer, ForeignKey("ambulances.id"))
    assigned_personnel = Column(JSON)  # Liste des IDs du personnel assigné
    
    # Timing
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    assigned_at = Column(DateTime(timezone=True))
    started_at = Column(DateTime(timezone=True))
    completed_at = Column(DateTime(timezone=True))
    estimated_duration = Column(Integer, default=30)  # minutes
    actual_duration = Column(Integer)  # minutes
    
    # Informations supplémentaires
    symptoms = Column(JSON)  # Liste des symptômes
    notes = Column(Text)
    
    # Relations
    hospital = relationship("Hospital")
    ambulance = relationship("Ambulance")

File: backend/models/Mission.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Mission = sequelize.define('Mission', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  patientName: {
    type: DataTypes.STRING(100),
    allowNull: false,
    field: 'patient_name'
  },
  patientPhone: {
    type: DataTypes.STRING(20),
    allowNull: false,
    field: 'patient_phone'
  },
  patientAge: {
    type: DataTypes.INTEGER,
    allowNull: true,
    field: 'patient_age'
  },
  patientCondition: {
    type: DataTypes.STRING(200),
    allowNull: false,
    field: 'patient_condition'
  },
  priority: {
    type: DataTypes.ENUM('critique', 'urgente', 'normale', 'faible'),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM('en_attente', 'assignee', 'en_cours', 'terminee', 'annulee'),
    allowNull: false,
    defaultValue: 'en_attente'
  },
  pickupAddress: {
    type: DataTypes.STRING(500),
    allowNull: false,
    field: 'pickup_address'
  },
  pickupLatitude: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'pickup_latitude'
  },
  pickupLongitude: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'pickup_longitude'
  },
  hospitalId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    field: 'hospital_id',
    references: {
      model: 'hospitals',
      key: 'id'
    }
  },
  ambulanceId: {
    type: DataTypes.INTEGER,
    allowNull: true,
    field: 'ambulance_id',
    references: {
      model: 'ambulances',
      key: 'id'
    }
  },
  assignedPersonnel: {
    type: DataTypes.JSON,
    allowNull: true,
    defaultValue: [],
    field: 'assigned_personnel'
  },
  assignedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    field: 'assigned_at'
  },
  startedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    field: 'started_at'
  },
  completedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    field: 'completed_at'
  },
  estimatedDuration: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 30,
    field: 'estimated_duration'
  },
  actualDuration: {
    type: DataTypes.INTEGER,
    allowNull: true,
    field: 'actual_duration'
  },
  symptoms: {
    type: DataTypes.JSON,
    allowNull: true,
    defaultValue: []
  },
  notes: {
    type: DataTypes.TEXT,
    allowNull: true
  }
}, {
  tableName: 'missions'
});

module.exports = Mission;